fiftyone/core/collections.py

```diff
@@ -9044,6 +9045,36 @@ def values(
         Returns:
             the list of values
         """
+
+        # If we do not need to follow insertion order, we can potentially use a
+        # covered index query to get the values directly from the index and
+        # avoid a COLLSCAN.
+        if not _enforce_natural_order:
+            field = None
+            if isinstance(field_or_expr, str):
+                field = field_or_expr
+            elif isinstance(field_or_expr, list):
+                field = field_or_expr[0]
+
+            # Checking _is_full_collection last as it could potentially be expensive.
+            # In the future, we might be less restrictive, but that needs
+            # further investigation.
+            if (
+                field in ["id", "_id", "filepath"]
+                and self._dataset.media_type != fom.GROUP
+                and self._is_full_collection()
```

Comment:
I think you had it right the first time, `_is_full_collection()` should handle the groups cases.
I don't see why it would be an expensive op, it's just a few comparisons.
```suggestion
                and self._is_full_collection()
```

Reply:
Is full collection returns true for grouped datasets. However the current implementation with values for groups only returns a slice. If we were to try to use this for id, the counts wouldn't match. Plus, we don't index by group name + id by default so there isn't any builtin guarantee optimization to preserve existing behavior

Reply:
oh interesting! glad to remove an unnecessary restriction here

Side note, I had checked:
```
ds.select_group_slices(ds.group_slices, _allow_mixed=True)._is_full_collection() # False
```
maybe an issue with _is_full_collection(), but not something to address here

Reply:
> Regarding _is_full_collection() as being potentially expensive, I recall Brian saying that _pipeline() could trigger off being frames/patches/etc collections or something and that's why it's also last in the method.

Yeah OK it should be safe though cuz of the other checks.